Update Vault docs re: faster role schema changes (#18057)
cockroachdb/docs: src/current/_includes/v23.2/sql/grant-revoke-schema-changes.md

User/role management operations (such as [`GRANT`]({% link {{ page.version.version }}/grant.md %}) and [`REVOKE`]({% link {{ page.version.version }}/revoke.md %})) are [schema changes]({% link {{ page.version.version }}/online-schema-changes.md %}). As such they inherit the [limitations of schema changes]({% link {{ page.version.version }}/online-schema-changes.md %}#limitations).

> For example, schema changes wait for concurrent [transactions]({% link {{ page.version.version }}/transactions.md %}) using the same resources as the schema changes to complete. In the case of [role memberships]({% link {{ page.version.version }}/security-reference/authorization.md %}#roles) being modified inside a transaction, most transactions need access to the set of role memberships. Using the default settings, any modifications to roles require schema leases to expire, which can take up to 5 minutes.

Author: Unfortunately the schema lease is a different kind of lease (terminology overload) that still needs documentation. It's related to how the database does name resolution and the caching of names of things

We have an issue to document this ([DOC-877](https://cockroachlabs.atlassian.net/browse/DOC-877)) which says in part that

> what we need is to explain to users that, in the query path, we need to resolve the names, but that the names are cached using leases that expire every 5 minutes. it’s pretty important to have this part documented, because it does come up from time to time when people run into problems if a node dies unexpectedly that is holding a lease - that can lead to some unexpected waits when doing a schema change on that object that was leased

and if you look at that issue you see a convo where it's explained "no this is a different lease" which .. is really unfortunate.  But hopefully I can work on DOC-877 Soon (tm).

If you want user/role management operations to finish more quickly, and do not care whether concurrent transactions will immediately see the side effects of those operations, set the [session variable]({% link {{ page.version.version }}/set-vars.md %}) `allow_role_memberships_to_change_during_transaction` to `true`.

When this session variable is enabled, any user/role management operations issued in the current session will only need to wait for the completion of statements in other sessions where `allow_role_memberships_to_change_during_transaction` is not enabled.

> 

Reviewer: Extra line break here.

Author: Ya that was intentional for skim-reading, so the final thing where we say "to speed this up, do X" stands out and is not at the tail end of a longish paragraph

> To accelerate user/role management operations across your entire application, you have the following options:

Reviewer: ```suggestion
To accelerate user/role management operations across your entire application, you can either:
```

Author: Ooh this one I'm not gonna take! I NEVER specify the number of options to follow in a list!!!!  Easier to maintain

1. You can set the session variable in all sessions by [passing it in the client connection string]({% link {{ page.version.version }}/connection-parameters.md %}#supported-options-parameters).

> 1. You can apply this setting globally to an entire cluster using the [`ALTER ROLE ALL`]({% link {{ page.version.version }}/alter-role.md %}) statement:

Reviewer: ```suggestion
1. Apply the `allow_role_memberships_to_change_during_transaction` setting globally to an entire cluster using the [`ALTER ROLE ALL`]({% link {{ page.version.version }}/alter-role.md %}) statement:
```

For example, schema changes wait for concurrent [transactions]({% link {{ page.version.version }}/transactions.md %}) using the same resources as the schema changes to complete. In the case of [role memberships]({% link {{ page.version.version }}/security-reference/authorization.md %}#roles) being modified inside a transaction, most transactions need access to the set of role memberships. Using the default settings, any modifications to roles require schema leases to expire, which can take up to 5 minutes.

This means that [long-running transactions]({% link {{ page.version.version }}/query-behavior-troubleshooting.md %}#hanging-or-stuck-queries) elsewhere in the system can cause user/role management operations inside transactions to take several minutes to complete. This can have a cascading effect. When a user/role management operation inside a transaction takes a long time to complete, it can in turn block all user-initiated transactions being run by your application, since the user/role management operation in the transaction has to commit before any other transactions that access role memberships (i.e., most transactions) can make progress.

> If you want user/role management operations to finish more quickly, and do not care whether concurrent transactions will immediately see the side-effects of those operations, set the [session variable]({% link {{ page.version.version }}/set-vars.md %}) `allow_role_memberships_to_change_during_transaction` to `true`.

Reviewer 2: Lets some how get an example to globally apply this to a cluster:
ALTER ROLE ALL SET allow_role_memberships_to_change_during_transaction = true;

Author: Good idea! I revised the end of this a bit to say (basically) you can either pass it on the conn string OR you can do ALTER ROLE ALL - PTAL and let me know what you think!
